src/modules/commander/commander.cpp

```diff
@@ -1761,31 +1768,29 @@ Commander::run()
 		orb_check(vtol_vehicle_status_sub, &updated);
 
 		if (updated) {
-			/* vtol status changed */
-			orb_copy(ORB_ID(vtol_vehicle_status), vtol_vehicle_status_sub, &vtol_status);
-			status.vtol_fw_permanent_stab = vtol_status.fw_permanent_stab;
-
-			/* Make sure that this is only adjusted if vehicle really is of type vtol */
-			if (is_vtol(&status)) {
+			const uint8_t prev_vtol_state = vtol_status.vtol_state;
+			if (orb_copy(ORB_ID(vtol_vehicle_status), vtol_vehicle_status_sub, &vtol_status) == PX4_OK) {
 
-				// Check if there has been any change while updating the flags
-				if (status.is_rotary_wing != vtol_status.vtol_in_rw_mode) {
-					status.is_rotary_wing = vtol_status.vtol_in_rw_mode;
-					status_changed = true;
-				}
+				status_flags.vtol_transition_failure = vtol_status.vtol_transition_failsafe;
 
-				if (status.in_transition_mode != vtol_status.vtol_in_trans_mode) {
-					status.in_transition_mode = vtol_status.vtol_in_trans_mode;
-					status_changed = true;
-				}
+				status.is_rotary_wing = true;
+				status.in_transition_mode = false;
+				status.in_transition_to_fw = false;
 
-				if (status.in_transition_to_fw != vtol_status.in_transition_to_fw) {
-					status.in_transition_to_fw = vtol_status.in_transition_to_fw;
-					status_changed = true;
+				switch (vtol_status.vtol_state) {
+				case vtol_vehicle_status_s::VEHICLE_VTOL_STATE_TRANSITION_TO_FW:
+				case vtol_vehicle_status_s::VEHICLE_VTOL_STATE_TRANSITION_TO_MC:
+					status.in_transition_mode = true;
+					break;
+				case vtol_vehicle_status_s::VEHICLE_VTOL_STATE_FW:
+					status.is_rotary_wing = false;
+					status.in_transition_to_fw = true;
+					break;
+				default:
+					break;
```

Comment:
Can't review this since I don't know the state machine on the other end well enough.